### PR TITLE
feat: support for camel cased table names and columns

### PR DESCRIPTION
**Pull request by mmahalwy**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,6 +48,9 @@ jobs:
             cd ../untagged-ids
             ./run.sh ../../migration-utils/build/index.js
             ./run.sh ../../codegen/build/index.js
+            cd ../camel-cased
```
Comment (stephenh) [inline .circleci/config.yml:51]: I'm pretty good with a new test directory/db; fwiw just talking out loud, I wonder if we could add a table or two to the existing `schema-misc` that had a few camel cased names?

**Comment by stephenh** [comment]
Looks like this existing `schema-misc` test is failing:

```
➤ YN0000:       Object {
➤ YN0000:     -   "authorId": Anything,
➤ YN0000:     +   "author": Object {
➤ YN0000:     +     "createdAt": 2022-06-01T04:16:45.256Z,
➤ YN0000:     +     "firstName": "firstName",
➤ YN0000:     +     "id": null,
➤ YN0000:     +     "lastName": null,
➤ YN0000:     +     "updatedAt": 2022-06-01T04:16:45.256Z,
➤ YN0000:     +   },
➤ YN0000:         "title": "title",
➤ YN0000:       }
➤ YN0000: 
➤ YN0000:        8 |     const b = newBook(em);
➤ YN0000:        9 |     // Then we did not set any updatedAt/createdAt values
➤ YN0000:     > 10 |     expect(b.__orm.data).toEqual({
➤ YN0000:          |                          ^
➤ YN0000:       11 |       authorId: expect.anything(),
➤ YN0000:       12 |       title: "title",
➤ YN0000:       13 |     });
```

Oh, but it's probably a good thing, b/c I think this PR actually fixes the `__orm.data.authorId` key to now be `__orm.data.author`. Nice!

**Review by stephenh — APPROVED** [review]
See comment musing if we should reuse `schema-misc` for this, but lgtm! Thanks!

File: .circleci/config.yml (hunk 1 above)
Comment (stephenh) [reply .circleci/config.yml:51]: Yep, that's what I was thinking. Granted, we don't have a great written-down heuristic for "when / when not to create a new testing directory"...kind of figuring it out as we go. :-)

**Comment by mmahalwy** [comment]
@stephenh updated

**Comment by stephenh** [comment]
@mmahalwy nice! Just looks like that one `Book.test.ts` is still failing.

**Comment by stephenh** [comment]
@mmahalwy ack! One more failure, later in the same test I think. :-)

**Comment by mmahalwy** [comment]
@stephenh hopefully this does it 😓

**Comment by stephenh** [comment]
There we go! :-D Merging! Thanks!

**Pull request merged**